.gitmodules

```diff
@@ -1,3 +1,6 @@
 [submodule "libs/azure-iot-middleware-freertos"]
 	path = libs/azure-iot-middleware-freertos
 	url = https://github.com/Azure/azure-iot-middleware-freertos.git
+[submodule "demos/projects/ESPRESSIF/esp32/components/esp-cryptoauthlib"]
```

Comment:
We decided to move this to CMake FetchContent to avoid downloading the new lib for samples not using it.

Reply:
Sounds good! Do I need to amend anything in my code?

Reply:
I was trying to see where to plug in the esp-cryptoauthlib as a CMake component. One finding is that the CMakeLists.txt for the ESPRESSIF projects (esp32, aziot) simply calls the project.cmake of the IDF.

Considering that the IDF CMake does all the steps (configure, generate, build...), I think the only solution would be one of the below two... what do you think @CIPop and @AdityaHPatwardhan ?

- <ins>**[Simplest]**</ins> Make esp-cryptoauthlib a component of the ESP-IDF itself rather than for one of the examples only... this removes the need to download it into the azure IoT middleware repo and may even reduce compatibility issues. I see this configuration happens here.
https://github.com/espressif/esp-idf/blob/master/.gitmodules#:~:text=%5Bsubmodule%20%22examples/peripherals,../../espressif/esp%2Dcryptoauthlib.git
- <ins>**[Slightly complicated]**</ins> Trigger a download of the esp-cryptoauthlib if someone selects the secure element option inside the esp-tls menuconfig. A good location for this can be this - https://github.com/espressif/esp-idf/blob/v4.4/components/esp-tls/CMakeLists.txt#:~:text=if(CONFIG_ESP_TLS_USE_SE),endif()

Do you recommend a better option?

Reply:
@CIPop Please ignore my comment above - I have figured out what I need to do to read the sdkconfig variable and conditionally fetch the esp-cryptoauthlib.